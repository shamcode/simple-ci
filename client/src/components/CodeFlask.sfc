<template>
    <textarea name={{name}} wrap="off" \style="display: none">{{value}}</textarea>
    <div class="code-flask" :ref="flaskContainer"></div>
</template>

<script>
    import Flask from 'codeflask';

    function CodeFlaskComponent( options, update, didMount ) {
        const name = $();
        const value = $();

        const state = options( {
            [ name ]: '',
            [ value ]: ''
        } );

        didMount( () => {
            const flask = new Flask( this.flaskContainer, {
                lineNumbers: true
            } );
            flask.updateCode( state[ value ] );
            flask.onUpdate(
                code => update( {
                    [ value ]: code
                } )
            );
        } );
    }

    export default Component( Template, CodeFlaskComponent );
</script>
