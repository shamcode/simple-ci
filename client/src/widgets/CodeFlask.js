import CodeFlask from 'codeflask';
import CodeFlaskTemplate from './CodeFlask.sht';
import { options } from 'sham-ui';

export default class CodeFlaskWidget extends CodeFlaskTemplate {
    @options get name() {
        return '';
    }
    @options get value() {
        return '';
    }

    @options
    afterRender() {
        this.flask = new CodeFlask( this.querySelector( 'div' ), {
            lineNumbers: true
        } );
        this.flask.onUpdate( ( code ) => {
            this.update( {
                value: code
            } );
        } );
    }

    update( data ) {
        super.update( ...arguments );
        if ( this.flask && data ) {
            this.flask.updateCode( data.value );
        }
    }
}
