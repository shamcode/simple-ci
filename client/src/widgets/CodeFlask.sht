<textarea name={{name}} wrap="off" style="display: none">{{value}}</textarea>
<div class="code-flask" :ref="flaskContainer"></div>
