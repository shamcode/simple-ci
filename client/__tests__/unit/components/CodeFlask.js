import CodeFlask from '../../../src/components/CodeFlask.sfc';
import renderer from 'sham-ui-test-helpers';
import * as directives from 'sham-ui-directives';

it( 'renders correctly', () => {
    const meta = renderer( CodeFlask, {
        directives
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
