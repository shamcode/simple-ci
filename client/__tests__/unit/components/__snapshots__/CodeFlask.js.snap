// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Constructor": "CodeFlaskComponent",
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "name": "",
    "value": "",
  },
  "html": "
<textarea wrap=\\"off\\" style=\\"display: none\\" name=\\"\\"></textarea>
<div class=\\"code-flask\\">
  <div class=\\"codeflask codeflask--has-line-numbers\\">
    <textarea class=\\"codeflask__textarea codeflask__flatten\\" spellcheck=\\"false\\" autocapitalize=\\"off\\" autocomplete=\\"off\\" autocorrect=\\"off\\"></textarea><pre class=\\"codeflask__pre codeflask__flatten  language-html\\"><code class=\\"codeflask__code  language-html\\"></code></pre>
    <div class=\\"codeflask__lines\\"><span class=\\"codeflask__lines__line\\">1</span></div>
  </div>
</div>
",
}
`;
